pkg/karmadactl/promote/promote.go

```diff
@@ -114,6 +114,14 @@ type CommandPromoteOption struct {
 	// DryRun tells if run the command in dry-run mode, without making any server requests.
 	DryRun bool
 
+	// AutoGeneratePropagationPolicy tells if the promote command create a PropagationPolicy automatically, otherwise one can create their own PropagationPolicy
+	AutoGeneratePropagationPolicy bool
+
+	// PropagationPolicyName is the name of a PropagationPolicy that automatically created by promote command,
+	// if not specified, a random suffix will be added to the PropagationPolicy.
+	// It is only valid when AutoGeneratePropagationPolicy is true.
+	PropagationPolicyName string
```

Comment:
If this policy already exists, what will happen?

Reply:
before building one policy it will [check] if the name exists (https://github.com/karmada-io/karmada/pull/3494/files/1be3cf421633636a171dc4129069a249fd0569e0#diff-6d515eefa1bc0f6f344292ee9f555c0005bc04d0905307bb67357cb234b43ca6R402)